webv2/table/utilities.go

```diff
@@ -0,0 +1,224 @@
+
+	interleaveTableSchema = createinterleaveTableSchema(interleaveTableSchema, table, tindex)
+
+	interleaveTableSchema = typeInterleaveColumn(interleaveTableSchema, table, columnId, colName, previoustype, updateType)
+	return interleaveTableSchema
+}
+
+func typeInterleaveColumn(interleaveTableSchema []InterleaveTableSchema, table, columnId, colName, previoustype, updateType string) []InterleaveTableSchema {
+
+	tindex := isTablePresent(interleaveTableSchema, table)
+	cindex := isColumnPresent(interleaveTableSchema[tindex].InterleaveColumnChanges, columnId)
+	interleaveTableSchema = createInterleaveColumntype(interleaveTableSchema, tindex, cindex, columnId, colName, previoustype, updateType)
+
+	if tindex != -1 && cindex != -1 {
+		interleaveTableSchema[tindex].InterleaveColumnChanges[cindex].ColumnId = columnId
+		interleaveTableSchema[tindex].InterleaveColumnChanges[cindex].ColumnName = colName
+		interleaveTableSchema[tindex].InterleaveColumnChanges[cindex].Type = previoustype
+		interleaveTableSchema[tindex].InterleaveColumnChanges[cindex].UpdateType = updateType
+
+	}
+
+	return interleaveTableSchema
+}
+
+func createInterleaveColumntype(interleaveTableSchema []InterleaveTableSchema, tindex int, cindex int, columnId string, colName string, previoustype string, updateType string) []InterleaveTableSchema {
+
+	if cindex == -1 {
+		if columnId != "" {
+			ic := InterleaveColumn{}
+			ic.ColumnId = columnId
+			ic.ColumnName = colName
+			ic.Type = previoustype
+			ic.UpdateType = updateType
+			interleaveTableSchema[tindex].InterleaveColumnChanges = append(interleaveTableSchema[tindex].InterleaveColumnChanges, ic)
+		}
+	}
+
+	return interleaveTableSchema
+}
+
+func trimRedundantInterleaveTableSchema(interleaveTableSchema []InterleaveTableSchema) []InterleaveTableSchema {
+	updatedInterleaveTableSchema := []InterleaveTableSchema{}
+	for _, v := range interleaveTableSchema {
+		if len(v.InterleaveColumnChanges) > 0 {
+			updatedInterleaveTableSchema = append(updatedInterleaveTableSchema, v)
+		}
+	}
+	return updatedInterleaveTableSchema
+}
+
+func updatedInterleaveTableSchema(conv *internal.Conv, interleaveTableSchema []InterleaveTableSchema) []InterleaveTableSchema {
+	for k, v := range interleaveTableSchema {
+		table := v.Table
+		for ind, col := range v.InterleaveColumnChanges {
+			if col.UpdateColumnName == "" {
+				interleaveTableSchema[k].InterleaveColumnChanges[ind].UpdateColumnName = col.ColumnName
+			}
+			if col.Type == "" {
+				interleaveTableSchema[k].InterleaveColumnChanges[ind].Type = conv.SpSchema[table].ColDefs[col.UpdateColumnName].T.Name
+			}
+			if col.UpdateType == "" {
+				interleaveTableSchema[k].InterleaveColumnChanges[ind].UpdateType = conv.SpSchema[table].ColDefs[col.UpdateColumnName].T.Name
+			}
+		}
+	}
+	return interleaveTableSchema
+}
+
+func UpdateNotNull(notNullChange, table, colName string, conv *internal.Conv) {
+
+	sp := conv.SpSchema[table]
+
+	switch notNullChange {
```

Comment:
Instead of using notNullChange as string better to use it as a boolean if it can have only two values and if not then use constants for these values

Reply:
Updated to used string constant. Please review it.